lambda-functions/cleanup-unused-images/main.go

```
@@ -0,0 +1,183 @@
+package main
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	log "github.com/sirupsen/logrus"
+)
+
+func main() {
+	log.SetLevel(log.DebugLevel)
+
+	lambda.Start(handler)
+}
+
+func handler() {
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(os.Getenv("AWS_REGION"))},
+	)
+
+	svc := ec2.New(sess)
+	uniqueUsedImages, err := getUniqueUsedImages(svc)
+	err = deleteAMIs(svc,uniqueUsedImages)
+
+	fmt.Println(uniqueUsedImages)
+
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			default:
+				log.WithError(aerr.Error("AWS Error."))
+			}
+		} else {
+			// Print the error, cast err to awserr.Error to get the Code and
+			// Message from an error.
+			log.WithError(err).Error("AWS Error.")
+		}
+		return
+	}
+}
+
+func deleteAMIs(svc *ec2.EC2, uniqueUsedImages []string) (error) {
+	imagesInput := &ec2.DescribeImagesInput{
+		Owners: []*string{
+			aws.String(os.Getenv("OWNER_ID")),
+		},
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("tag:Name"),
+				Values: []*string{aws.String("mattermost-cloud-*")},
+			},
+		},
+	}
+	snapshots, err := getAllSnapshots(os.Getenv("OWNER_ID"), svc)
+	allImages, err := svc.DescribeImages(imagesInput)
+	oldImages, err := filterImagesByDateRange(allImages.Images, 730)
+	dryRun := true
+	for _, i := range oldImages {
+		imageForCleanup := contains(uniqueUsedImages, *i.ImageId)
+		fmt.Println(imageForCleanup)
+		fmt.Println(*i.ImageId + ": De-registering AMI named \"" + *i.Name + "\"...")
+		cleanupImageInput := &ec2.DeregisterImageInput{
+		ImageId: &imageForCleanup,
+		DryRun: &dryRun,
+		}
+		cleanupImage, _ := svc.DeregisterImage(cleanupImageInput)
```

Are we ignoring an error here? Looks like we should be handling it.